### PR TITLE
Introduced new lib for MieleCloudService

### PR DESCRIPTION
**Pull request by Grizzelbee**
Introduced new lib for MieleCloudService to detect Miele devices regardless if they are connected via XGW 3000 gateway or directly via WLAN (WifiConn@ct). So more devices can be detected with this lib.

### CODE REVIEW

**Comment by Grizzelbee** [comment]
Push!

File: lib/adapters/mielecloudservice.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    if ( (device._source !== 'ip') && (device._source !== 'mdns') ) {
+        return callback(null, false, ip);
+    }
+
+    let detected   = false;
+    let deviceName = 'Miele Appliance';
+
+    if (device._source === 'ip') {
+        tools.httpGet('http://' + ip + '/homebus', (err, data) => {
+            if (!err && data && ( reIsMieleXGW3000.test(data) ) ){
```
Comment (Apollon77) [inline lib/adapters/mielecloudservice.js:23]: This code is executed async ... so ip detection will never work ...

Comment (Apollon77) [reply lib/adapters/mielecloudservice.js:23]: Best isto make the "if detected" code into  a separate function and call from both relevant places ... first wins. so set callback to "null" after called and check efore calling callback if not null

Comment (Grizzelbee) [reply lib/adapters/mielecloudservice.js:23]: The IP-code is copied from Miele-Adapter, as I don't own a XGW3000 gateway.
I'll try to find someone owning one and fix the code - detection of the gateway itself (hopefully via mdns) willl be suffiicent and is maybe already working.
In worst case gateway devices respond to the miele-adapter detection and WIFI-devices respond to MieleCloudService.

**Pull request merged**